### PR TITLE
chore: use rayon in pq transform and remove async calls

### PR DESCRIPTION
**Pull request by eddyxu**

### CODE REVIEW

**Review by BubbleCal — APPROVED** [review]

File: rust/lance-index/src/vector/pq.rs [hunk 1]
```diff
@@ -333,89 +333,64 @@ where
...
+                    message: format!(
+                        "Expect to be a float vector array, got: {:?}",
+                        fsl.value_type()
+                    ),
+                    location: location!(),
+                })?;
+
+        let values = flatten_data
+            .as_slice()
+            .par_chunks(dim)
```
Comment (chebbyChefNEQ) [inline rust/lance-index/src/vector/pq.rs:355]: nit: would it be more effcient to do `.chunk_exact(dim). into_par_iter().map(...)`?

**Review by chebbyChefNEQ — APPROVED** [review]
one nit + would you happen to have benchmark numbers?

File: rust/lance-index/src/vector/pq.rs (hunk 1 above)
Comment (eddyxu) [reply rust/lance-index/src/vector/pq.rs:355]: It does not compile in that way

```
error[E0599]: the method `into_par_iter` exists for struct `ChunksExact<'_, <T as ArrowFloatType>::Native>`, but its trait bounds were not satisfied
    --> rust/lance-index/src/vector/pq.rs:356:14
     |
353  |           let values = flatten_data
     |  ______________________-
354  | |             .as_slice()
355  | |             .chunks_exact(dim)
356  | |             .into_par_iter()
     | |_____________-^^^^^^^^^^^^^
     |
    ::: /Users/lei/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/slice/iter.rs:1828:1
     |
1828 |   pub struct ChunksExact<'a, T: 'a> {
     |   --------------------------------- doesn't satisfy `_: IntoParallelIterator` or `_: ParallelIterator`
     |
     = note: the following trait bounds were not satisfied:
             `std::slice::ChunksExact<'_, <T as lance_arrow::ArrowFloatType>::Native>: rayon::iter::ParallelIterator`
             which is required by `std::slice::ChunksExact<'_, <T as lance_arrow::ArrowFloatType>::Native>: rayon::iter::IntoParallelIterator`
             `&std::slice::ChunksExact<'_, <T as lance_arrow::ArrowFloatType>::Native>: rayon::iter::ParallelIterator`
             which is required by `&std::slice::ChunksExact<'_, <T as lance_arrow::ArrowFloatType>::Native>: rayon::iter::IntoParallelIterator`
             `&mut std::slice::ChunksExact<'_, <T as lance_arrow::ArrowFloatType>::Native>: rayon::iter::ParallelIterator`
             which is required by `&mut std::slice::ChunksExact<'_, <T as lance_arrow::ArrowFloatType>::Native>: rayon::iter::IntoParallelIterator`

warning: unused import: `rayon::prelude`
  --> rust/lance-index/src/vector/pq.rs:18:5
```

**Comment by eddyxu** [comment]
> one nit + would you happen to have benchmark numbers?
did not find a handy benchmark for PQ::transform yet, could be next time.

**Pull request merged**